### PR TITLE
Sihp data

### PR DESCRIPTION
**Pull request by Radonirinaunimi**
This adds the single hadron production datasets:
| Dataset names  | paper  | hepdata  |
|---|---|---|
|  PHENIX-2009_SHP_200GEV_PI0_ALL | [paper](https://inspirehep.net/literature/1282448)  | [data](https://www.hepdata.net/record/ins1282448)  |
|  STAR-2006_SHP_200GEV_PI0_ALL | [paper](https://inspirehep.net/literature/1253360)  |  [data](https://www.hepdata.net/record/ins1253360) |
| PHENIX-2013_SHP_510GEV_PI0_ALL  | [paper](https://inspirehep.net/literature/1396712)  |  [data](https://www.hepdata.net/record/ins1396712) |
| STAR-2013_SHP_510GEV_PI0_ALL-LOWRAP  | [paper](https://inspirehep.net/literature/1674826)  |  [data](https://www.hepdata.net/record/ins1674826) |
| STAR-2013_SHP_510GEV_PI0_ALL-HIGHRAP  | [paper](https://inspirehep.net/literature/1674826)  |  [data](https://www.hepdata.net/record/ins1674826) |

These do not have the corresponding FK tables yet but we do have the codes to generate the predictions [NNPDF/sihp-pp](https://github.com/NNPDF/sihp-pp).

### CODE REVIEW

File: nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/metadata.yaml
Comment (scarlehoff) [inline nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/metadata.yaml:51]:
```suggestion
```

Don't add a theory section if there is no theory yet, otherwise the code will (understandably) try to use the null.pineappl.lz4 fktable you mentioned here.

File: nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/metadata.yaml
Comment (scarlehoff) [inline nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/metadata.yaml:22]:
```suggestion
      label: "$A_{LL}$"
```

File: nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/metadata.yaml
Comment (scarlehoff) [inline nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/metadata.yaml:24]: please add a process to the process option library

**Review by scarlehoff — COMMENTED** [review]
Please, rebase on top of master after I merge #2153

File: nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/metadata.yaml
Comment (Radonirinaunimi) [reply nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/metadata.yaml]: Done in [5af0d56](https://github.com/NNPDF/nnpdf/pull/2154/commits/5af0d563f28adaa864d3620182bb7c7d58c88e77).

File: nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/metadata.yaml
Comment (Radonirinaunimi) [reply nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/metadata.yaml]: Done in [5af0d56](https://github.com/NNPDF/nnpdf/pull/2154/commits/5af0d563f28adaa864d3620182bb7c7d58c88e77).

File: nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/metadata.yaml
Comment (Radonirinaunimi) [reply nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/metadata.yaml:24]: Yes, yes. I was planning to do so but then more things came up and pushed this down to the list of priorities ^^U

**Comment by t7phy** [comment]
@Radonirinaunimi given that the naming convention is <exp.name>_<process>_<energy>_<misc.info>, I think it would better to have years in the <misc.info> section instead of renaming the experiments to include the year.

**Comment by scarlehoff** [comment]
It depends whether the different years should be taken as separated experiments (when using per-experiment grouping for instance). So with `STAR_SHIP_2006_200GEV_ALL` all STAR would be grouped together, while `STAR-2006_SHIP_200GEV_ALL` groups the `STAR-2006` dataset together.

(btw, for a moment I thought "Why do we have an 'all' observable, what does that mean?" 😆 )

File: nnpdf_data/nnpdf_data/new_commondata/PHENIX-2009_SHP_200GEV/uncertainties.yaml
Comment (giacomomagni) [inline nnpdf_data/nnpdf_data/new_commondata/PHENIX-2009_SHP_200GEV/uncertainties.yaml:10]: By looking at the caption of Table VI of https://arxiv.org/pdf/1402.6296, I think we need to add the lumi uncertainty: 4.2 * 10^-4, correlated additive. ( while we are not considering the polarization scale uncertainty, 4.8%).

File: nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/uncertainties.yaml
Comment (giacomomagni) [inline nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/uncertainties.yaml:6]: From https://arxiv.org/pdf/1309.1800 fig 7, I think we need to add 6% polarisation beam uncertainty: CORR, STAR2006POL.

Comment (Radonirinaunimi) [reply nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/uncertainties.yaml:6]: Upon first reading the caption and the corresponding text, I understood that the 6% scaling was already included in the vertical bar and hence in the overall systematics. But you might be right that this is not the case.

It is a bit annoying that these experimental papers are not always clear in stating what they did.

File: nnpdf_data/nnpdf_data/new_commondata/PHENIX-2009_SHP_200GEV/uncertainties.yaml
Comment (Radonirinaunimi) [reply nnpdf_data/nnpdf_data/new_commondata/PHENIX-2009_SHP_200GEV/uncertainties.yaml:10]: Why additive?

Comment (giacomomagni) [reply nnpdf_data/nnpdf_data/new_commondata/PHENIX-2009_SHP_200GEV/uncertainties.yaml:10]: I think we treat the lumi unc as additive (?)

Comment (Radonirinaunimi) [reply nnpdf_data/nnpdf_data/new_commondata/PHENIX-2009_SHP_200GEV/uncertainties.yaml:10]: Usually, unless explicitly stated otherwise, luminosity uncs are treated as multiplicative.

File: nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/uncertainties.yaml
Comment (giacomomagni) [reply nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/uncertainties.yaml:6]: Indeed this is not super clear. For instance in the first bin If I do 0.06*0.023=0.00138, while the quoted syst is 0.02. 
So I'm not sure the paper says:

```
Uncertainty in the product of beam polarizations results in a 6% vertical scale uncertainty as
indicated in the figure.
```

**Comment by giacomomagni** [comment]
I think these measurements might be also relevant (cc @enocera):

STAR 510 GeV: https://inspirehep.net/literature/1674826
PHENIX 510 GeV: https://inspirehep.net/literature/1789851

Charged hadrons (just for reference):
PHENIX 200 GeV inclusive: https://inspirehep.net/literature/1089402
PHENIX 200 GeV pions: https://inspirehep.net/literature/1315330

**Comment by Radonirinaunimi** [comment]
> I think these measurements might be also relevant (cc @enocera):
> 
> STAR 510 GeV: https://inspirehep.net/literature/1674826 
> PHENIX 510 GeV: https://inspirehep.net/literature/1789851
> 
I've implemented them. The reference for PHENIX was not the correct one (it contains the charged measurements) and I rectified it in the table above. These measurements, AFAIU, were performed in ~2012-2013 but I just took 2013 for all of them.

> Charged hadrons (just for reference): PHENIX 200 GeV inclusive: https://inspirehep.net/literature/1089402 PHENIX 200 GeV pions: https://inspirehep.net/literature/1315330

Given that we'll not include this in the paper, I will not implement them now but rather do so once we have interfaced https://github.com/NNPDF/sihp-pp.

**Comment by Radonirinaunimi** [comment]
@scarlehoff please check that [7301318](https://github.com/NNPDF/nnpdf/pull/2154/commits/73013184519754de5e3b92c7f975b5e8f85171f0) address all the required changes in terms of having new process options.

And with this, modulo reviews, this PR is done AFAIC.

File: nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/uncertainties.yaml
Comment (Radonirinaunimi) [reply nnpdf_data/nnpdf_data/new_commondata/STAR-2006_SHP_200GEV/uncertainties.yaml]: I included the 6% beam polarization in any case - better be conservative about them if we are not sure.

**Review by giacomomagni — APPROVED** [review]

**Review by scarlehoff — COMMENTED** [review]
Thanks @Radonirinaunimi 
do you want to wait until the grids are ready to merge it? (so that it is merged "completed" with the right grid names even if the fktables are not in the theory yet)

**Comment by Radonirinaunimi** [comment]
I think it'd be best to merge this as implementing the PineAPPL interface to produce the grids might take some time, mainly because we'll have to fix the C and Fortran APIs once `V1`-format is ready.

**Comment by t7phy** [comment]
@Radonirinaunimi before merging, could you please add the identified hadron name somewhere in the dataset name or observable name (wherever you prefer). Having the PI0 mentioned explcitly in the names and not having to look in the metadata description would be super helpful especially when moving towards FF fits and preparing runcards.

**Comment by Radonirinaunimi** [comment]
> @Radonirinaunimi before merging, could you please add the identified hadron name somewhere in the dataset name or observable name (wherever you prefer). Having the PI0 mentioned explcitly in the names and not having to look in the metadata description would be super helpful especially when moving towards FF fits and preparing runcards.

Good point! Especially since for some experiments/measurements we have different final state hadrons. To conform with the other dataset namings (such as gauge boson production) I added the type and charge to the dataset name.

**Comment by t7phy** [comment]
> Good point! Especially since for some experiments/measurements we have different final state hadrons. To conform with the other dataset namings (such as gauge boson production) I added the type and charge to the dataset name.

Thanks!

**Pull request merged**
